### PR TITLE
Misc cleanups

### PR DESCRIPTION
**Pull request by glasserc**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -202,4 +202,52 @@ again with the command::
...
+.. code-block:: bash
+
+   $ git checkout master
+   $ git pull
+   $ release
+   $ postrelease
+
+Then:
+
+The "Back to development" commit cannot be pushed to master because we don't allow pushes to master.
```
Comment (leplatrem) [inline README.rst]: There might be some zest.releaser options possible to avoid that (in setup.cfg IDK)

File: README.rst [hunk 2]
```diff
@@ -202,4 +202,52 @@ again with the command::
...
+- When the PR is approved, merge it using bors
+
+Then:
+
+.. code-block:: bash
+
+   $ git checkout master
+   $ git pull
+   $ release
+   $ postrelease
```
Comment (leplatrem) [inline README.rst]: Add a note telling this will be deployed to STAGE at Mozilla?

File: README.rst [hunk 3]
```diff
@@ -202,4 +202,52 @@ again with the command::
...
+   $ git reset --hard origin/master
+
+Then:
+
+- Open another PR
+
+Then:
+
+- Create a release on the Github page using the contents of the CHANGELOG as the body
+- Open a Bugzilla bug telling ops to deploy the new release
```
Comment (leplatrem) [inline README.rst]: in PROD

File: README.rst (hunk 2 above)
Comment (glasserc) [reply README.rst]: This is only true for remote-settings I think.

Comment (glasserc) [reply README.rst]: But it's easy to add a note that clarifies that if you want!

File: README.rst (hunk 3 above)
Comment (glasserc) [reply README.rst]: Again, this depends on the project.

File: README.rst (hunk 1 above)
Comment (glasserc) [reply README.rst]: There might be, but nothing easy. Anyhow, we could easily tell developers to `git reset --hard HEAD^` to get rid of it. But the absence of a changelog stub kind of bothers me.

File: README.rst (hunk 2 above)
Comment (leplatrem) [reply README.rst]: That's fine for me

**Pull request merged**